Bug located in test using python 3.13

### 🔍 Before submitting the issue

- [x] I have searched among the existing issues
- [x] I am using a Python virtual environment

### 🐞 Description of the bug

FAILED tests/test_post.py::test_vector_object - AttributeError: 'PyLocalContainer' object has no attribute '_local_collection'
FAILED tests/test_post.py::test_surface_object - AttributeError: 'PyLocalContainer' object has no attribute '_local_collection'

Both these tests are shipped now for the CI to pass. Please update accordingly once the issue is fixed.

### 📝 Steps to reproduce

Run test_vector_object and test_surface_object using python 3.13 in Windows github container. 

### 💻 Which operating system are you using?

Windows

### 📀 Which ANSYS version are you using?

24R2 (this bug is independent of Fluent server).

### 🐍 Which Python version are you using?

3.13

### 📦 Installed packages

```shell
.
```

--- a/.github/workflows/ci_cd.yml
+++ b/.github/workflows/ci_cd.yml
@@ -23,7 +23,7 @@ env:
   # It applies 7 days retention policy by default.
   RESET_EXAMPLES_CACHE: 0
   ANSYSLMD_LICENSE_FILE: ${{ format('1055@{0}', secrets.LICENSE_SERVER) }}
-  MAIN_PYTHON_VERSION: "3.11"
+  MAIN_PYTHON_VERSION: "3.12"
   LIBRARY_NAME: 'ansys-fluent-visualization'
   PYFLUENT_TIMEOUT_FORCE_EXIT: 60
   PYFLUENT_LAUNCH_CONTAINER: 1
@@ -72,7 +72,7 @@ jobs:
     strategy:
       matrix:
         os: [windows-latest, public-ubuntu-latest-8-cores]
-        python-version: ['3.10', '3.11', '3.12', '3.13']
+        python-version: ['3.10', '3.11', '3.12', '3.13', '3.14']
       fail-fast: false
 
     steps:

--- a/.github/workflows/nightly-doc-build.yml
+++ b/.github/workflows/nightly-doc-build.yml
@@ -11,7 +11,7 @@ permissions:
 
 env:
   DOCUMENTATION_CNAME: 'visualization.fluent.docs.pyansys.com'
-  MAIN_PYTHON_VERSION: '3.11'
+  MAIN_PYTHON_VERSION: '3.12'
   DOC_DEPLOYMENT_IMAGE_TAG: v25.2.3
   ANSYSLMD_LICENSE_FILE: ${{ format('1055@{0}', secrets.LICENSE_SERVER) }}
   PYFLUENT_TIMEOUT_FORCE_EXIT: 60

--- a/.github/workflows/nightly-test-run.yml
+++ b/.github/workflows/nightly-test-run.yml
@@ -20,7 +20,7 @@ jobs:
     strategy:
       matrix:
         os: [windows-latest, public-ubuntu-latest-8-cores]
-        python-version: ['3.10', '3.11', '3.12', '3.13']
+        python-version: ['3.10', '3.11', '3.12', '3.13', '3.14']
       fail-fast: false
 
     steps:

--- a/README.rst
+++ b/README.rst
@@ -58,7 +58,7 @@ the project support team, email `pyansys.core@ansys.com <pyansys.core@ansys.com>
 Installation
 ------------
 The ``ansys-fluent-visualization`` package supports Python 3.10 through Python
-3.13 on Windows and Linux.
+3.14 on Windows and Linux.
 
 If you are using Python 3.10, download and install the wheel file for the ``vtk`` package from
 `here for Windows <https://github.com/pyvista/pyvista-wheels/raw/main/vtk-9.1.0.dev0-cp310-cp310-win_amd64.whl>`_

--- a/doc/source/getting_started/index.rst
+++ b/doc/source/getting_started/index.rst
@@ -14,7 +14,7 @@ page on the Ansys website.
 Install package
 ***************
 The ``ansys-fluent-visualization`` package supports Python 3.10 through
-Python 3.13 on Windows and Linux.
+Python 3.14 on Windows and Linux.
 
 Install the latest release from `PyPi
 <https://pypi.org/project/ansys-fluent-visualization/>`_ with:

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -23,7 +23,7 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.10,<3.14"
+python = ">=3.10,<=3.14"
 importlib-metadata = {version = "^4.0", python = "<3.9"}
 ansys-fluent-core = "~=0.39.dev0"
 pyvista = ">=0.44.0"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -23,7 +23,7 @@ packages = [
 ]
 
 [tool.poetry.dependencies]
-python = ">=3.10,<=3.14"
+python = ">=3.10,<3.15"
 importlib-metadata = {version = "^4.0", python = "<3.9"}
 ansys-fluent-core = "~=0.39.dev0"
 pyvista = ">=0.44.0"

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -29,7 +29,7 @@ ansys-fluent-core = "~=0.39.dev0"
 pyvista = ">=0.44.0"
 matplotlib = ">=3.6.0"
 pyvistaqt = { version = "~=0.11.1", optional = true }
-pyside6 = { version = "~=6.8.1", optional = true }
+pyside6 = { version = ">=6.8.1,<7.0.0", optional = true }
 
 [tool.poetry.extras]
 interactive = ["pyvistaqt"]

--- a/src/ansys/fluent/interface/post_objects/meta.py
+++ b/src/ansys/fluent/interface/post_objects/meta.py
@@ -718,10 +718,10 @@ class PyLocalContainer(MutableMapping):
 
     def __init__(self, parent, object_class, api_helper, name=""):
         """Initialize the 'PyLocalContainer' object."""
+        self._local_collection = {}
         self._parent = parent
         self._name = name
         self.__object_class = object_class
-        self._local_collection = {}
         self.__api_helper = api_helper
         self.type = "named-object"
         self._command_names = []

--- a/src/ansys/fluent/interface/post_objects/meta.py
+++ b/src/ansys/fluent/interface/post_objects/meta.py
@@ -271,7 +271,7 @@ class PyLocalPropertyMeta(PyLocalBaseMeta):
             self._api_helper = api_helper(self)
             self._parent = parent
             self._on_change_cbs = []
-            annotations = self.__class__.__dict__.get("__annotations__")
+            annotations = getattr(self.__class__, "__annotations__", {}) or {}
             if isinstance(getattr(self.__class__, "value", None), property):
                 value_annotation = annotations.get("_value")
             else:

--- a/src/ansys/fluent/visualization/renderer.py
+++ b/src/ansys/fluent/visualization/renderer.py
@@ -90,8 +90,9 @@ class GraphicsWindow:
         """
         self._list_of_positions.append(position)
         if isinstance(graphics_obj._obj, GraphicsDefn):
-            locals()["object"] = locals().pop("graphics_obj")
-            self._graphics_objs.append({**locals()})
+            local_vars = locals().copy()
+            local_vars["object"] = local_vars.pop("graphics_obj")
+            self._graphics_objs.append(local_vars)
         else:
             warnings.warn("Only graphics objects are supported.")
 
@@ -115,8 +116,9 @@ class GraphicsWindow:
         """
         self._list_of_positions.append(position)
         if isinstance(plot_obj._obj, PlotDefn):
-            locals()["object"] = locals().pop("plot_obj")
-            self._graphics_objs.append({**locals()})
+            local_vars = locals().copy()
+            local_vars["object"] = local_vars.pop("plot_obj")
+            self._graphics_objs.append(local_vars)
         else:
             warnings.warn("Only 2D plot objects are supported.")
 

--- a/tests/test_visualization.py
+++ b/tests/test_visualization.py
@@ -31,11 +31,6 @@ from ansys.fluent.interface.post_objects.post_object_definitions import ContourD
 # Adding this makes sure that these tests only run in ubuntu CI.
 if sys.platform == "win32":
     pytest.skip("Skipping this module on Windows", allow_module_level=True)
-else:
-    # These tests are failing in python 3.13. Investigate it along with
-    # "test_vector_object" and "test_surface_object" in "test_post.py".
-    if sys.version_info[:2] == (3, 13):
-        pytest.skip("Skipped on Python 3.13", allow_module_level=True)
 
 from unittest.mock import patch
 

